moveit_core/robot_state/test/test_cartesian_planner.cpp

```diff
@@ -0,0 +1,387 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2013, Willow Garage, Inc.
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Willow Garage nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Author: Ioan Sucan */
+#include <moveit_resources/config.h>
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <moveit/cartesian_planner/cartesian_planner.h>
+#include <urdf_parser/urdf_parser.h>
+#include <gtest/gtest.h>
+#include <sstream>
+#include <algorithm>
+#include <ctype.h>
+
+class OneRobot : public testing::Test
+{
+protected:
+  void SetUp() override
+  {
+    static const std::string MODEL2 =
```

Comment:
Can you use the new the new testing framework for this?
https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html

Reply:
I think that is a great idea but I don't think I will have time and would rather get this merged with the tests as is.

Reply:
@mlautman when feature requests like this are asked for that you don't have time for, I think an ideal is to add an inline TODO

Reply:
@rhaschke I looked into it but I found this issue when using the new framework. https://github.com/ros-planning/moveit/issues/1386

Reply:
I thought of using `RobotModelBuilder` as for example [here](https://github.com/ros-planning/moveit/blob/a20b3632ef74ff57028a16ff066cb007ab1c1a42/moveit_core/robot_model/test/test.cpp#L88-L99). You don't need a full-fledged Panda robot for this test.

Reply:
Thanks for looking into this, @mlautman. If you don't have time, leave it as is.